### DISCUSSION TITLE
Setting `RenderLayers` for the grids

### DISCUSSION THREAD

**Discussion by musjj** [Trouv/bevy_ecs_ldtk#390] [Q&A]
I'm using `bevy_ecs_ldtk` and I'm trying to set the render layer (for offscreen camera purposes) for the spawned grids, but I'm having trouble making it work. I tried setting it in the bundle:

```rust
#[derive(Component, Default)]
pub struct Wall;

#[derive(Bundle, LdtkIntCell)]
pub struct WallBundle {
    wall: Wall,
    layer: RenderLayers,
}

impl Default for GroundBundle {
    fn default() -> Self {
        Self {
            wall: Wall,
            layer: RenderLayers::layer(1),
        }
    }
}
```

But it doesn't work. I also tried putting it alongside the world bundle:

```rust
commands.spawn((
    LdtkWorldBundle {
        ldtk_handle: asset_server.load("tilemap.ldtk"),
        ..default()
    },
    RenderLayers::layer(1),
));
```

But it's also not working. Is there a proper way of doing this?

**Comment by musjj** [comment]
Figured out a solution:

```rust
fn add_layer(
    mut commands: Commands,
    wall_query: Query<&Parent, Added<Wall>>,
    parent_query: Query<Entity, (With<Parent>, Without<Wall>)>,
) {
    for parent in ground_query.iter() {
        if let Ok(entity) = parent_query.get(parent.get()) {
            commands.entity(entity).insert(RenderLayers::layer(1));
        }
    }
}
```

But this doesn't affect the background. I'm not sure how to target the background entity directly, so I'm currently using this hacky background:

```rust
fn fix_background(
    mut commands: Commands,
    query: Query<Entity, (With<Sprite>, Without<...>)>,
) {
    for sprite in query.iter() {
        commands.entity(sprite).insert(RenderLayers::layer(1));
    }
}
```

I had to litter it with `Without<...>` so that it doesn't affect any unwanted sprites. Is there a better way of doing this?

**Comment by Trouv** [comment]
Sorry I'm aware this issue is very old.

I'm guessing it's a `bevy_ecs_tilemap` thing. I think the component needs to be applied to the tilemap entity rather than the tile entities. Just poking around w/ an inspector, it seems tiles have a `bevy_ecs_tilemap`-native component `TileVisible` for visibility, rather than bevy's `Visibility`/`ComputedVisibility`, which I believe the `RenderLayers` machinery leverages?

Setting it on the tilemap is what you're doing, it looks like, it's just that you're using parentage to get there. Nowadays, you can get to it directly by querying for `LayerMetadata` components.
